components/sup/src/cli.rs

```diff
@@ -44,6 +105,224 @@ mod test {
+                .into_temp_path();
+            let key_file_str = key_file.to_str().unwrap();
+            let cmd_str = format!("hab-sup run --key {}", key_file_str);
+            let cmd_vec = Vec::from_iter(cmd_str.split_whitespace());
+            assert!(cli().get_matches_from_safe(cmd_vec).is_err());
+        }
+
+        #[test]
+        fn cert_file_requires_key_file() {
+            let cert_file = NamedTempFile::new()
+                .expect("Failed to create temporary cert_file")
+                .into_temp_path();
+            let cert_file_str = cert_file.to_str().unwrap();
+            let cmd_str = format!("hab-sup run --certs {}", cert_file_str);
+            let cmd_vec = Vec::from_iter(cmd_str.split_whitespace());
+            assert!(cli().get_matches_from_safe(cmd_vec).is_err());
+        }
+
+        assert_cli_cmd!(should_take_pkg_or_artifact_as_arg,
+                        "hab-sup run core/redis",
+                        "PKG_IDENT_OR_ARTIFACT" => "core/redis");
+
+        assert_cli_cmd!(should_handle_application_and_environment,
+                        "hab-sup run --application foobar --environment fizbang",
+                        "APPLICATION" => "foobar",
+                        "ENVIRONMENT" => "fizbang");
+
+        assert_cli_cmd!(application_and_environment_should_have_short_flags,
+                        "hab-sup run -a foobar -e fizbang",
+                        "APPLICATION" => "foobar",
+                        "ENVIRONMENT" => "fizbang");
+
+        #[test]
+        fn application_requires_environment() {
+            let cmd_vec = Vec::from_iter("hab-sup run -a foobar".split_whitespace());
+            assert!(cli().get_matches_from_safe(cmd_vec).is_err());
+        }
+
+        #[test]
+        fn environment_requires_application() {
+            let cmd_vec = Vec::from_iter("hab-sup run -e fizbang".split_whitespace());
+            assert!(cli().get_matches_from_safe(cmd_vec).is_err());
+        }
+
+        assert_cli_cmd!(should_handle_group,
+                        "hab-sup run --group groupy",
+                        "GROUP" => "groupy");
+
+        assert_cli_cmd!(group_should_have_default_value,
+                        "hab-sup run",
+                        "GROUP" => "default");
+
+        assert_cli_cmd!(should_handle_standalone_topology,
+                        "hab-sup run --topology standalone",
+                        "TOPOLOGY" => "standalone");
+
+        assert_cli_cmd!(should_handle_leader_topology,
+                        "hab-sup run --topology leader",
+                        "TOPOLOGY" => "leader");
+
+        assert_cli_cmd!(topology_should_have_a_short_flag,
+                        "hab-sup run -t standalone",
+                        "TOPOLOGY" => "standalone");
+
+        #[test]
+        fn topology_should_fail_when_a_bad_value_is_passed() {
+            let cmd_vec = Vec::from_iter("hab-sup run --topology foobar".split_whitespace());
```

Comment:
A more likely wrong value would be if someone got `--topology` confused with strategy, so I'd test `at-once` and  `rolling`. Similarly (but reversed) for `--strategy`.